shared/src/api/index.coffee

```diff
@@ -165,13 +165,16 @@ class APIHandlerBase
 
     onSuccess ?= DEFAULT_SUCCESS
     onFail ?= DEFAULT_FAIL
-    return new Promise (resolve) =>
+    return new Promise (resolve, reject) =>
+
       delay =>
         @_xhr.request(requestConfig)
+          .then(((resp) -> resolve(resp); resp), ((resp) -> reject(resp); Promise.reject(resp)))
```

Comment:
Oh and if you want to test it, mark all your courses as non-preview, then add a new teacher without any courses.